examples/list_consumer_groups.c

```diff
@@ -184,24 +187,29 @@ int64_t parse_int(const char *what, const char *str) {
 static void
 cmd_list_consumer_groups(rd_kafka_conf_t *conf, int argc, char **argv) {
         rd_kafka_t *rk;
-        const char **states_str = NULL;
         char errstr[512];
         rd_kafka_AdminOptions_t *options;
         rd_kafka_event_t *event = NULL;
         rd_kafka_error_t *error = NULL;
         int i;
-        int retval     = 0;
-        int states_cnt = 0;
+        int retval          = 0;
+        int states_cnt      = 0;
+        int group_types_cnt = 0;
         rd_kafka_consumer_group_state_t *states;
+        rd_kafka_consumer_group_type_t *group_types;
 
-
-        if (argc >= 1) {
-                states_str = (const char **)&argv[0];
-                states_cnt = argc;
-        }
+        states_cnt = parse_int("state count", argv[0]);
         states = calloc(states_cnt, sizeof(rd_kafka_consumer_group_state_t));
         for (i = 0; i < states_cnt; i++) {
-                states[i] = parse_int("state code", states_str[i]);
+                states[i] = parse_int("state code", argv[i + 1]);
+        }
+        group_types_cnt = parse_int("group type count", argv[states_cnt + 1]);
```

Comment:
If you call it without arguments it gives a segfault, so let's validate number of arguments.

You can validate number of arguments parsing first count of states and types and then if number of arguments is correct proceed to parsing the state and group type values. In case validation on number of arguments fails you call the usage function and terminate the example.

Reply:
Okay.